Description: Don't use the DOCK hint on the login window as this stops it from getting keyboard focus
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=598235
Bug-Ubuntu: https://bugs.launchpad.net/gdm/+bug/447049 

Index: gdm-2.30.2/gui/simple-greeter/gdm-greeter-login-window.c
===================================================================
--- gdm-2.30.2.orig/gui/simple-greeter/gdm-greeter-login-window.c	2010-06-22 13:10:49.934144597 +1000
+++ gdm-2.30.2/gui/simple-greeter/gdm-greeter-login-window.c	2010-06-22 13:11:26.264150018 +1000
@@ -1543,7 +1543,7 @@
         gtk_window_set_skip_taskbar_hint (GTK_WINDOW (login_window), TRUE);
         gtk_window_set_skip_pager_hint (GTK_WINDOW (login_window), TRUE);
         gtk_window_stick (GTK_WINDOW (login_window));
-        gtk_window_set_type_hint (GTK_WINDOW (login_window), GDK_WINDOW_TYPE_HINT_DOCK);
+        /*gtk_window_set_type_hint (GTK_WINDOW (login_window), GDK_WINDOW_TYPE_HINT_DOCK);*/ /* Docks cannot get focus */
         gtk_container_set_border_width (GTK_CONTAINER (login_window), 0);
 
         g_signal_connect (login_window,
